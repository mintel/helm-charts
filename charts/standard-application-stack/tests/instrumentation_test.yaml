suite: Test opentelemetry-operator Instrumentation
templates:
  - instrumentation.yaml
  - deployment.yaml
release:
  namespace: test-namespace
tests:
  - it: creates an Instrumentation using Python as an example
    template: instrumentation.yaml
    set:
      global.clusterEnv: qa
      global.cloudProvider.accountId: mock-account-id
      global.cloudProvider.region: mock-aws-region
      global.clusterName: mock-cluster-name
      otelInstrumentation.injectPython.enabled: true
      otelInstrumentation.env:
        - name: OTEL_ENV_VAR
          value: mock-env-var-value
      otelInstrumentation.exporter.endpoint: "mock-endpoint"
      otelInstrumentation.resource.resourceAttributes.mock-resource-key: mock-resource-value
      otelInstrumentation.resource.addK8sUIDAttributes: true
      otelInstrumentation.propagators: [mock-propagators]
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - hasDocuments:
          count: 1
      - isKind:
          of: Instrumentation
      - equal:
          path: spec.exporter.endpoint
          value: "mock-endpoint"
      - contains:
          path: spec.env
          content:
            name: OTEL_ENV_VAR
            value: mock-env-var-value
      - equal:
          path: spec.resource.resourceAttributes
          value:
            cloud.account.id: mock-account-id
            cloud.platform: aws_eks
            cloud.provider: aws
            cloud.region: mock-aws-region
            deployment.environment: qa
            k8s.cluster.name: mock-cluster-name
            mock-resource-key: mock-resource-value
            service.name: example-app
            service.namespace: test-namespace
      - equal:
          path: spec.resource.addK8sUIDAttributes
          value: true
      - equal:
          path: spec.propagators
          value: [mock-propagators]

  - it: adds the env var injection annotation to the Deployment
    set:
      global.clusterEnv: qa
      global.name: example-app
      otelInstrumentation.injectPython.enabled: true
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - template: instrumentation.yaml
        equal:
          path: metadata.namespace
          value: "test-namespace"
      - template: instrumentation.yaml
        equal:
          path: metadata.name
          value: "example-app"
      - template: deployment.yaml
        equal:
          path: spec.template.metadata.annotations["instrumentation.opentelemetry.io/inject-sdk"]
          value: "test-namespace/example-app"

  - it: adds the Python injection annotation to the Deployment
    set:
      global.clusterEnv: qa
      global.name: example-app
      otelInstrumentation.injectPython.enabled: true
      otelInstrumentation.autoInjectSidecar.enabled: true
      otelInstrumentation.injectPython.env:
        - name: OTEL_ENV_VAR
          value: mock-env-var-value
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - template: instrumentation.yaml
        equal:
          path: metadata.namespace
          value: "test-namespace"
      - template: instrumentation.yaml
        equal:
          path: metadata.name
          value: "example-app"
      - template: deployment.yaml
        equal:
          path: spec.template.metadata.annotations["instrumentation.opentelemetry.io/inject-python"]
          value: "test-namespace/example-app"
      - template: deployment.yaml
        isNull:
          path: spec.template.metadata.annotations["instrumentation.opentelemetry.io/inject-sdk"]

  - it: adds the Java injection annotation to the Deployment
    set:
      global.clusterEnv: qa
      global.name: example-app
      otelInstrumentation.injectJava.enabled: true
      otelInstrumentation.autoInjectSidecar.enabled: true
      otelInstrumentation.injectJava.env:
        - name: OTEL_ENV_VAR
          value: mock-env-var-value
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - template: instrumentation.yaml
        equal:
          path: metadata.namespace
          value: "test-namespace"
      - template: instrumentation.yaml
        equal:
          path: metadata.name
          value: "example-app"
      - template: deployment.yaml
        equal:
          path: spec.template.metadata.annotations["instrumentation.opentelemetry.io/inject-java"]
          value: "test-namespace/example-app"
      - template: deployment.yaml
        isNull:
          path: spec.template.metadata.annotations["instrumentation.opentelemetry.io/inject-sdk"]

  - it: adds the NodeJS injection annotation to the Deployment
    set:
      global.clusterEnv: qa
      global.name: example-app
      otelInstrumentation.injectNodeJS.enabled: true
      otelInstrumentation.autoInjectSidecar.enabled: true
      otelInstrumentation.injectNodeJS.env:
        - name: OTEL_ENV_VAR
          value: mock-env-var-value
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - template: instrumentation.yaml
        equal:
          path: metadata.namespace
          value: "test-namespace"
      - template: instrumentation.yaml
        equal:
          path: metadata.name
          value: "example-app"
      - template: deployment.yaml
        equal:
          path: spec.template.metadata.annotations["instrumentation.opentelemetry.io/inject-nodejs"]
          value: "test-namespace/example-app"
      - template: deployment.yaml
        isNull:
          path: spec.template.metadata.annotations["instrumentation.opentelemetry.io/inject-sdk"]

  - it: adds the .NET injection annotation to the Deployment
    set:
      global.clusterEnv: qa
      global.name: example-app
      otelInstrumentation.injectDotNet.enabled: true
      otelInstrumentation.autoInjectSidecar.enabled: true
      otelInstrumentation.injectDotNet.env:
        - name: OTEL_ENV_VAR
          value: mock-env-var-value
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - template: instrumentation.yaml
        equal:
          path: metadata.namespace
          value: "test-namespace"
      - template: instrumentation.yaml
        equal:
          path: metadata.name
          value: "example-app"
      - template: deployment.yaml
        equal:
          path: spec.template.metadata.annotations["instrumentation.opentelemetry.io/inject-dotnet"]
          value: "test-namespace/example-app"
      - template: deployment.yaml
        isNull:
          path: spec.template.metadata.annotations["instrumentation.opentelemetry.io/inject-sdk"]

  - it: sets the container names annotation
    template: deployment.yaml
    set:
      global.clusterEnv: qa
      global.name: example-app
      otelInstrumentation.autoInjectSidecar.enabled: true
      otelInstrumentation.injectPython.enabled: true
      otelInstrumentation.containerNames:
        - foo
        - bar
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - equal:
          path: spec.template.metadata.annotations["instrumentation.opentelemetry.io/container-names"]
          value: "bar,foo"
