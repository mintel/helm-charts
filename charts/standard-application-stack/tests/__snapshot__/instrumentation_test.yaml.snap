adds the .NET injection annotation to the Deployment:
  1: |
    apiVersion: opentelemetry.io/v1alpha1
    kind: Instrumentation
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-20"
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      dotnet:
        env:
          - name: OTEL_ENV_VAR
            value: mock-env-var-value
      exporter:
        endpoint: http://grafana-agent.monitoring.svc.cluster.local:4317
      resource:
        resourceAttributes:
          cloud.platform: aws_eks
          cloud.provider: aws
          deployment.environment: qa
          service.name: example-app
          service.namespace: test-namespace
      sampler:
        type: parentbased_always_on
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        helm.sh/chart: standard-application-stack-3.61.3
        secret.reloader.stakater.com/reload: example-app-app
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: example-app
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          annotations:
            instrumentation.opentelemetry.io/inject-dotnet: test-namespace/example-app
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: example-app
            app.mintel.com/env: qa
            app.mintel.com/region: ${CLUSTER_REGION}
            name: example-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: qa
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: example-app-app
              image: registry.gitlab.com/test:auto-replaced
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          imagePullSecrets:
            - name: image-pull-gitlab
            - name: image-pull-docker-hub
          securityContext:
            runAsUser: 1000
          serviceAccountName: example-app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: example-app
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
          volumes: null
adds the Java injection annotation to the Deployment:
  1: |
    apiVersion: opentelemetry.io/v1alpha1
    kind: Instrumentation
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-20"
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      exporter:
        endpoint: http://grafana-agent.monitoring.svc.cluster.local:4317
      java:
        env:
          - name: OTEL_ENV_VAR
            value: mock-env-var-value
      resource:
        resourceAttributes:
          cloud.platform: aws_eks
          cloud.provider: aws
          deployment.environment: qa
          service.name: example-app
          service.namespace: test-namespace
      sampler:
        type: parentbased_always_on
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        helm.sh/chart: standard-application-stack-3.61.3
        secret.reloader.stakater.com/reload: example-app-app
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: example-app
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          annotations:
            instrumentation.opentelemetry.io/inject-java: test-namespace/example-app
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: example-app
            app.mintel.com/env: qa
            app.mintel.com/region: ${CLUSTER_REGION}
            name: example-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: qa
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: example-app-app
              image: registry.gitlab.com/test:auto-replaced
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          imagePullSecrets:
            - name: image-pull-gitlab
            - name: image-pull-docker-hub
          securityContext:
            runAsUser: 1000
          serviceAccountName: example-app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: example-app
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
          volumes: null
adds the NodeJS injection annotation to the Deployment:
  1: |
    apiVersion: opentelemetry.io/v1alpha1
    kind: Instrumentation
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-20"
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      exporter:
        endpoint: http://grafana-agent.monitoring.svc.cluster.local:4317
      nodejs:
        env:
          - name: OTEL_ENV_VAR
            value: mock-env-var-value
      resource:
        resourceAttributes:
          cloud.platform: aws_eks
          cloud.provider: aws
          deployment.environment: qa
          service.name: example-app
          service.namespace: test-namespace
      sampler:
        type: parentbased_always_on
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        helm.sh/chart: standard-application-stack-3.61.3
        secret.reloader.stakater.com/reload: example-app-app
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: example-app
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          annotations:
            instrumentation.opentelemetry.io/inject-nodejs: test-namespace/example-app
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: example-app
            app.mintel.com/env: qa
            app.mintel.com/region: ${CLUSTER_REGION}
            name: example-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: qa
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: example-app-app
              image: registry.gitlab.com/test:auto-replaced
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          imagePullSecrets:
            - name: image-pull-gitlab
            - name: image-pull-docker-hub
          securityContext:
            runAsUser: 1000
          serviceAccountName: example-app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: example-app
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
          volumes: null
adds the Python injection annotation to the Deployment:
  1: |
    apiVersion: opentelemetry.io/v1alpha1
    kind: Instrumentation
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-20"
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      exporter:
        endpoint: http://grafana-agent.monitoring.svc.cluster.local:4317
      python:
        env:
          - name: OTEL_ENV_VAR
            value: mock-env-var-value
      resource:
        resourceAttributes:
          cloud.platform: aws_eks
          cloud.provider: aws
          deployment.environment: qa
          service.name: example-app
          service.namespace: test-namespace
      sampler:
        type: parentbased_always_on
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        helm.sh/chart: standard-application-stack-3.61.3
        secret.reloader.stakater.com/reload: example-app-app
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: example-app
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          annotations:
            instrumentation.opentelemetry.io/inject-python: test-namespace/example-app
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: example-app
            app.mintel.com/env: qa
            app.mintel.com/region: ${CLUSTER_REGION}
            name: example-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: qa
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: example-app-app
              image: registry.gitlab.com/test:auto-replaced
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          imagePullSecrets:
            - name: image-pull-gitlab
            - name: image-pull-docker-hub
          securityContext:
            runAsUser: 1000
          serviceAccountName: example-app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: example-app
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
          volumes: null
adds the env var injection annotation to the Deployment:
  1: |
    apiVersion: opentelemetry.io/v1alpha1
    kind: Instrumentation
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-20"
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      exporter:
        endpoint: http://grafana-agent.monitoring.svc.cluster.local:4317
      resource:
        resourceAttributes:
          cloud.platform: aws_eks
          cloud.provider: aws
          deployment.environment: qa
          service.name: example-app
          service.namespace: test-namespace
      sampler:
        type: parentbased_always_on
  2: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        helm.sh/chart: standard-application-stack-3.61.3
        secret.reloader.stakater.com/reload: example-app-app
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: example-app
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          annotations:
            instrumentation.opentelemetry.io/inject-sdk: test-namespace/example-app
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: example-app
            app.mintel.com/env: qa
            app.mintel.com/region: ${CLUSTER_REGION}
            name: example-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: qa
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: example-app-app
              image: registry.gitlab.com/test:auto-replaced
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          imagePullSecrets:
            - name: image-pull-gitlab
            - name: image-pull-docker-hub
          securityContext:
            runAsUser: 1000
          serviceAccountName: example-app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: example-app
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
          volumes: null
creates an Instrumentation:
  1: |
    apiVersion: opentelemetry.io/v1alpha1
    kind: Instrumentation
    metadata:
      annotations:
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-20"
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      env:
        - name: OTEL_ENV_VAR
          value: mock-env-var-value
      exporter:
        endpoint: mock-endpoint
      propagators:
        - mock-propagators
      resource:
        addK8sUIDAttributes: true
        resourceAttributes:
          cloud.account.id: mock-account-id
          cloud.platform: aws_eks
          cloud.provider: aws
          cloud.region: mock-aws-region
          deployment.environment: qa
          k8s.cluster.name: mock-cluster-name
          mock-resource-key: mock-resource-value
          service.name: example-app
          service.namespace: test-namespace
      sampler:
        type: parentbased_always_on
sets the container names annotation:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        helm.sh/chart: standard-application-stack-3.61.3
        secret.reloader.stakater.com/reload: example-app-app
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: example-app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: example-app
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          annotations:
            instrumentation.opentelemetry.io/container-names: bar,foo
            instrumentation.opentelemetry.io/inject-sdk: test-namespace/example-app
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: example-app
            app.mintel.com/env: qa
            app.mintel.com/region: ${CLUSTER_REGION}
            name: example-app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: qa
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: example-app-app
              image: registry.gitlab.com/test:auto-replaced
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 0
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          imagePullSecrets:
            - name: image-pull-gitlab
            - name: image-pull-docker-hub
          securityContext:
            runAsUser: 1000
          serviceAccountName: example-app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: example-app
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
          volumes: null
